core/src/main/java/com/linecorp/armeria/common/CookieBuilder.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.BitSet;
+
+import javax.annotation.Nullable;
+
+/**
+ * Builds a {@link Cookie}.
+ */
+public final class CookieBuilder {
+
+    private static final BitSet VALID_COOKIE_ATTRIBUTE_VALUE_OCTETS = validCookieAttributeValueOctets();
+
+    // path-value        = <any CHAR except CTLs or ";">
+    private static BitSet validCookieAttributeValueOctets() {
+        final BitSet bits = new BitSet();
+        for (int i = 32; i < 127; i++) {
+            bits.set(i);
+        }
+        bits.set(';', false);
+        return bits;
+    }
+
+    private static int firstInvalidOctet(CharSequence cs, BitSet bits) {
```

Comment:
I saw the same method in `CookieUtil`. Can we reuse it?

Reply:
Fixed